sentry-types/src/protocol/envelope.rs

```diff
@@ -282,13 +294,19 @@ impl Envelope {
                     writeln!(writer)?;
                     continue;
                 }
+                EnvelopeItem::Profile(profile) => {
+                    profile.to_writer(&mut writer)?;
+                    writeln!(writer)?;
+                    continue;
```

Comment:
If you don’t need the extra header properties (which I think you don’t), you can use the same code here and below:

`serde_json::to_writer`, combined with the match down below for the `item_type`.